src/pairinteraction_gui/config/ket_config.py

```diff
@@ -285,48 +258,44 @@ class QnSQDT(QnBase):
         self.s = s
+        self.items = {}
         super().__init__(parent)
 
     def setupWidget(self) -> None:
-        spin_boxes = self._spin_boxes = {}
-
-        spin_boxes["n"] = IntSpinBox(self, vmin=1, vdefault=80, tooltip="Principal quantum number n")
-
-        spin_boxes["l"] = IntSpinBox(self, tooltip="Orbital angular momentum l")
+        self.items["n"] = QnItemInt(self, "n", vmin=1, vdefault=80, tooltip="Principal quantum number n")
+        self.items["l"] = QnItemInt(self, "l", tooltip="Orbital angular momentum l")
 
         s = self.s
         if s % 1 == 0:
-            spin_boxes["j"] = IntSpinBox(self, vmin=int(s), vdefault=int(s), tooltip="Total angular momentum j")
-            spin_boxes["m"] = IntSpinBox(self, vmin=-999, vdefault=0, tooltip="Magnetic quantum number m")
+            self.items["j"] = QnItemInt(self, "j", vmin=int(s), vdefault=int(s), tooltip="Total angular momentum j")
+            self.items["m"] = QnItemInt(self, "m", vmin=-999, vdefault=0, tooltip="Magnetic quantum number m")
         else:
-            spin_boxes["j"] = HalfIntSpinBox(self, vmin=s, vdefault=s, tooltip="Total angular momentum j")
-            spin_boxes["m"] = HalfIntSpinBox(self, vmin=-999, vdefault=0.5, tooltip="Magnetic quantum number m")
+            self.items["j"] = QnItemHalfInt(self, "j", vmin=s, vdefault=s, tooltip="Total angular momentum j")
+            self.items["m"] = QnItemHalfInt(self, "m", vmin=-999, vdefault=0.5, tooltip="Magnetic quantum number m")
 
 
 class QnMQDT(QnBase):
     """Configuration for earth alkali atoms using MQDT."""
 
-    default_deactivated: ClassVar = ["l_ryd"]
-
     def __init__(self, parent: Optional[QWidget] = None, m_is_int: bool = False) -> None:
         self.m_is_int = m_is_int
+        self.items = {}
         super().__init__(parent)
 
     def setupWidget(self) -> None:
-        spin_boxes = self._spin_boxes = {}
-        spin_boxes["nu"] = DoubleSpinBox(
-            self, vmin=1, vdefault=80, vstep=1, tooltip="Effective principal quantum number nu"
+        self.items["nu"] = QnItemDouble(
+            self, "nu", vmin=1, vdefault=80, vstep=1, tooltip="Effective principal quantum number nu"
         )
-        spin_boxes["s"] = DoubleSpinBox(self, vmin=0, vmax=1, vstep=0.1, tooltip="Spin s")
-        spin_boxes["j"] = DoubleSpinBox(self, vstep=1, tooltip="Total angular momentum j")
+        self.items["s"] = QnItemDouble(self, "s", vmin=0, vmax=1, vstep=0.1, tooltip="Spin s")
+        self.items["j"] = QnItemDouble(self, "j", vstep=1, tooltip="Total angular momentum j")
 
         if self.m_is_int:
-            spin_boxes["f"] = IntSpinBox(self, vmin=0, vdefault=0, tooltip="Total angular momentum f")
-            spin_boxes["m"] = IntSpinBox(self, vmin=-999, vdefault=0, tooltip="Magnetic quantum number m")
+            self.items["f"] = QnItemInt(self, "f", vmin=0, vdefault=0, tooltip="Total angular momentum f")
```

Comment:
I would find "QnSpinBoxInt" or "SpinBoxQnInt" more descriptive because such a name would tell me immediately that this is a spin box object in the GUI. But I do not have a strong feeling. If item fits better to the names of other objects, I would also be fine with item.

Reply:
Ok, I just saw that you also have a "RangeItem" and items are in general more than just a SpinBoxes.